code/clue.js

```diff
@@ -1,59 +1,350 @@
-const randomSelector = array => {
-  return array[Math.floor(Math.random() * array.length)]
-}
+
+const randomSelector = (array) => {
+  return array[Math.floor(Math.random() * array.length)];
+};
 
 // CREATE AN OBJECT THAT KEEPS THE MYSTERY.
 // With a killer, a weapon and a room.
 // The values will be set later.
+const mystery = {
+  killer: undefined,
+  weapon: undefined,
+  room: undefined,
+  favWeapon: undefined,
+};
+
+// shuffleFavouriteWeapon function
+const duplicateItems = (arr) => {
+  let newArr = [];
+  for (let i = 0; i < 10; i++) {
+    newArr = newArr.concat(arr);
+  }
+  return newArr;
+};
+
+const favWeapons = [rope, knife, candlestick, dumbbell, poison, axe];
+const shuffleFavouriteWeapon = () => {
+  mystery.favWeapon = shuffle(favWeapons, weapons);
+  document.getElementById(
+    "mysteryName"
+  ).innerHTML = `Favorite wepon is ${mystery.favWeapon.id}`;
+};
+
+const shuffle = (favArr, arr) => {
+  const favDuplicated = duplicateItems(favArr);
+  const concatArr = [...arr, ...favDuplicated];
+  return randomSelector(concatArr);
+};
+// ----------------WEAPON AND ROOM
+
+const shuffleFavRoom = () => {
+  if (mystery.weapon !== undefined) {
+    const roomList = weaponRooms[mystery.weapon.id];
+    mystery.room = shuffle(roomList, rooms);
+    document.getElementById("roomName").innerHTML = mystery.room;
+  } else {
+    mystery.room = randomSelector(rooms);
+    document.getElementById("roomName").innerHTML = mystery.room;
+  }
+};
 
 // This function will be invoked when you click on the killer card.
 const pickKiller = () => {
   // This will randomly select a killer from the suspects. And add that to the mystery object.
-  mystery.killer = randomSelector(suspects)
+  mystery.killer = randomSelector(suspects);
 
   // This will change the background color of the card to the one connected to the chosen killer and show the full name of the killer. Feel free to add more things to show about the killer.
-  document.getElementById('killerCard').style.background = mystery.killer.color
+
+  document.getElementById("killerCard").style.background = mystery.killer.color;
+
+  document.getElementById("killerName").innerHTML =
+    mystery.killer.firstName + " " + mystery.killer.lastName;
+
+  document.getElementById("killerOccupation").innerHTML =
+    mystery.killer.occupation;
+
   document.getElementById(
-    'killerName'
-  ).innerHTML = `${mystery.killer.firstName} ${mystery.killer.lastName}`
-}
+    "killerImage"
+  ).innerHTML = `<img src="${mystery.killer.image}">`;
+};
 
 // CREATE FUNCTIONS pickWeapon and pickRoom in a similar way.
 
+const pickWeapon = () => {
+  mystery.weapon = randomSelector(weapons);
+  document.getElementById(
+    "weaponName"
+  ).innerHTML = `${mystery.weapon.name} which weight is ${mystery.weapon.weight}.`;
+};
+
+// LOADER
+
+document.getElementById("loaderItem").style.display = "none";
+
+// Invoke function
+
+document.getElementById("killerCard").onclick = () => {
+  loader();
```

Comment:
It's a bit confusing that even after you pick single card (suspect, weapon or room), spinner is still spinning - it could be stopped until next selection